src/core_modules/capture-core/components/DataEntries/SingleEventRegistrationEntry/DataEntryWrapper/DataEntry/epics/newEventDataEntry.epics.js

```diff
@@ -87,6 +87,11 @@ export const openNewEventInDataEntryEpic = (action$: InputObservable, store: Red
             return page === 'new';
         }),
         filter((action) => {
+            const { programId } = store.value.currentSelections;
+            const program = programCollection.get(programId);
```

Comment:
Hey, 
I think that this logic is already in `getEventProgramThrowIfNotFound` and the function can be called here. WDYT?
Thanks!

Reply:
Thanks for the comment. In this code block the program can be either tracker or event and it doesn't feel right to throw if it is a tracker program (i.e., it's not an exception to get a tracker program here). I would reserve exception throwing for cases where we didn't get what we expected.

(and yes this code needs some refactoring also, but that's for another day)